build/generate-docs.ts

```diff
@@ -511,9 +511,37 @@ function createExpressionsContent() {
             }
             content += `\n### ${key}\n`;
             content += `${value.doc}\n`;
-            value.example.syntax.method.unshift(`"${key}"`);
-            content += `\nSyntax:\n${codeBlockMarkdown(`[${value.example.syntax.method.join(', ')}]: ${value.example.syntax.result}`, 'js')}\n`;
-            content += `\nExample:\n${codeBlockMarkdown(`"some-property": ${formatJSON(value.example.value)}`)}\n`;
+
+            content += '\nSyntax:\n';
+            const syntax: {
+                variants: {
+                    parameters: string[];
+                    'output-type': string;
+                }[];
+                parameters?: {
+                    name: string;
+                    type?: string;
+                    description?: string;
+                }[];
+            } = value.syntax;
+            const codeBlockLines = syntax.variants.map((variant) => {
+                return `[${[`"${key}"`, ...variant.parameters].join(', ')}]: ${variant['output-type']}`;
+            });
+            content += `${codeBlockMarkdown(codeBlockLines.join('\n'), 'js')}\n`;
+            const parameterLines = (syntax.parameters ?? []).map((param) => {
```

Comment:
Consider a simple for look instead of a map call since the conversion is to a line that later on gets added, so a for feels a bit more readable, to me at least.